src/main/java/com/hillel/items_exchange/dto/UserDto.java

```diff
@@ -27,17 +29,17 @@
     private String email;
     @NotNull(message = "{invalid.not-null}")
     private Boolean online;
-    @NotNull(message = "{invalid.not-null}")
     @Size(min = 2, max = 50, message = "{invalid.size}")
     private String firstName;
-    @NotNull(message = "{invalid.not-null}")
     @Size(min = 2, max = 50, message = "{invalid.size}")
     private String lastName;
-    @NotNull(message = "{invalid.not-null}")
     private String avatarImage;
     @NotNull(message = "{invalid.not-null}")
     @PastOrPresent(message = "{invalid.past-or-present.date}")
     @JsonFormat(pattern = "yyyy-MM-dd")
     private LocalDate lastOnlineTime;
-
+    @NotNull(message = "{invalid.not-null}")
```

Comment:
Hm.. Why user's lastName or avatarImage can be null, but phone or children can't?

Reply:
I followed db constraints when added those validations. In case with children or phone, we have an empty array. But I will try to deal with null and may remove those validation from children and phones.

Reply:
Need to check this moment with @rpkyrych. It's just business logic, but I think this entities may not be nulls.

Reply:
All of those NotNull constraints make a user to add/change all his information. Right after the user registration avatarImage, firstName, lastName are empty string. So if he wants add first name only (actually change value of firstName from "" to user first name), he won't be able to do it because of lastName and avatarImage can't be "".
According to Declaration of thsi task I have to give a possibility to add only what a user wants and edit it if he wants. 
So I am going to create a custom annotation to have "" is valid, null isn't. Am I on the rigth way?

Reply:
Discussed internally - @lubchik-14 will additionally check the problem.